lib/client/signalflow/message_router.js

```diff
@@ -80,11 +80,17 @@ function getRoutedMessageHandler(params, onMessage) {
             numBatchesDetermined = true;
             flushBuffer();
           }
+
           onMessage(msg);
           break;
         case 'metadata':
         case 'event':
+          onMessage(msg);
+          break;
         case 'control-message':
+          if (isRetryPatchMode && !numBatchesDetermined) {
```

Comment:
As discussed, this is not correct. You'll end up suppressing end of channel and abort control messages.